sdk/core/azure-core-management/src/main/java/com/azure/core/management/Region.java

```diff
@@ -0,0 +1,363 @@
+    /**
+     * Germany Northeast
+     */
+    public static final Region GERMANY_NORTHEAST = new Region("germanynortheast", "Germany Northeast");
+    /*
+     * Azure Government Cloud
+     */
+    /**
+     * U.S. government cloud in Virginia.
+     */
+    public static final Region GOV_US_VIRGINIA = new Region("usgovvirginia", "US Gov Virginia");
+
+    /**
+     * U.S. government cloud in Iowa.
+     */
+    public static final Region GOV_US_IOWA = new Region("usgoviowa", "US Gov Iowa");
+
+    /**
+     * U.S. government cloud in Arizona.
+     */
+    public static final Region GOV_US_ARIZONA = new Region("usgovarizona", "US Gov Arizona");
+
+    /**
+     * U.S. government cloud in Texas.
+     */
+    public static final Region GOV_US_TEXAS = new Region("usgovtexas", "US Gov Texas");
+
+    /**
+     * U.S. Department of Defense cloud - East.
+     */
+    public static final Region GOV_US_DOD_EAST = new Region("usdodeast", "US DoD East");
+
+    /**
+     * U.S. Department of Defense cloud - Central.
+     */
+    public static final Region GOV_US_DOD_CENTRAL = new Region("usdodcentral", "US DoD Central");
+
+    private final String name;
+    private final String label;
+
+    /**
+     * @return predefined Azure regions
+     */
+    public static Region[] values() {
+        Collection<Region> valuesCollection = VALUES_BY_NAME.values();
+        return valuesCollection.toArray(new Region[valuesCollection.size()]);
+    }
+
+    private Region(String name, String label) {
+        this.name = name;
+        this.label = label;
+        VALUES_BY_NAME.put(name.toLowerCase(Locale.ROOT), this);
+    }
+
+    /**
+     * Creates a region from a name and a label.
+     *
+     * @param name the uniquely identifiable name of the region
+     * @param label the label of the region
+     * @return the newly created region
+     */
+    public static Region create(String name, String label) {
+        Region region = VALUES_BY_NAME.get(name.toLowerCase(Locale.ROOT));
+        if (region != null) {
+            return region;
+        } else {
+            return new Region(name, label);
+        }
+    }
+
+    @JsonValue
+    @Override
+    public String toString() {
+        return name();
+    }
+
+    /**
+     * @return the name of the region
+     */
+    public String name() {
+        return this.name;
+    }
+
+    /**
+     * @return the label of the region
+     */
+    public String label() {
+        return this.label;
+    }
+
+    /**
+     * Finds a region based on a label or name.
+     * <p>
+     * A region name is lower-cased label with spaces removed.
+     *
+     * @param labelOrName the region name or label
+     * @return the found region or null if there's no such region
+     */
+    public static Region findByLabelOrName(String labelOrName) {
+        if (labelOrName == null) {
+            return null;
+        }
+
+        return VALUES_BY_NAME.get(labelOrName.toLowerCase(Locale.ROOT).replace(" ", ""));
+    }
```

Comment:
This method likely can be removed (SDK can have a util method to do this via `values()` method).